### PR TITLE
Add Storage KMS samples.

### PR DESCRIPTION
**Pull request by jsimonweb**

### CODE REVIEW

File: storage/api/Storage/Storage.cs [hunk 1]
```diff
@@ -498,6 +511,55 @@ private void ViewBucketIamMembers(string bucketName)
         }
         // [END remove_bucket_iam_member]
 
+        // [START storage_set_bucket_default_kms_key]
+        private void AddBucketDefaultKmsKey(string bucketName,
+            string keyLocation, string kmsKeyRing, string kmsKeyName)
+        {
+            string KeyLocation = keyLocation;
```
Comment (frankyn) [inline storage/api/Storage/Storage.cs]: is this line necessary?

File: storage/api/Storage/Storage.cs [hunk 2]
```diff
@@ -498,6 +511,55 @@ private void ViewBucketIamMembers(string bucketName)
...
+            });
+        }
+        // [END storage_set_bucket_default_kms_key]
+
+        // [START storage_upload_with_kms_key]
+        private void UploadEncryptedFileWithKmsKey(string bucketName,
+            string keyLocation, string kmsKeyRing, string kmsKeyName,
+            string localPath, string objectName = null)
+        {
+            string KeyLocation = keyLocation;
```
Comment (frankyn) [inline storage/api/Storage/Storage.cs]: Same comment about this being line being necessary?

File: storage/api/StorageTest/StorageTest.cs [hunk 3]
```diff
@@ -597,6 +645,8 @@ public void TestEncryptUploadAndDownload()
                 File.ReadAllText("Hello-downloaded.txt"));
         }
 
+
```
Comment (frankyn) [inline storage/api/StorageTest/StorageTest.cs]: white space may be accidental.

File: storage/api/StorageTest/StorageTest.cs [hunk 4]
```diff
@@ -558,6 +588,24 @@ public void TestAddBucketIamMember()
             Assert.DoesNotContain(member, printedIamMembers.Stdout);
         }
 
+        [Fact]
+        public void TestAddBucketDefaultKmsKey()
+        {
+            var addBucketDefaultKmsKeyResponse =
+                Run("add-bucket-default-kms-key", _bucketName1,
+                    s_kmsKeyLocation, _kmsKeyRing, _kmsKeyName);
+            AssertSucceeded(addBucketDefaultKmsKeyResponse);
```
Comment (frankyn) [inline storage/api/StorageTest/StorageTest.cs]: Does this test only verify the sample STDOUT response and not the state of the bucket default kms key?

Comment (jsimonweb) [reply storage/api/StorageTest/StorageTest.cs]: Updated to confirm the bucket's default kms key.

File: storage/api/StorageTest/StorageTest.cs [hunk 5]
```diff
@@ -558,6 +588,26 @@ public void TestAddBucketIamMember()
...
+            Assert.Contains(_kmsKeyName, bucketMetadata.Stdout);
+        }
+
+        [Fact]
+        public void TestUploadWithKmsKey()
+        {
+            var uploadWithKmsKeyResponse = Run("upload-with-kms-key",
+                    _bucketName1, s_kmsKeyLocation, _kmsKeyRing,
+                    _kmsKeyName, CollectRegionalObject("Hello.txt"));
+            AssertSucceeded(uploadWithKmsKeyResponse);
```
Comment (frankyn) [inline storage/api/StorageTest/StorageTest.cs]: @jsimonweb could you confirm the object kms key name as well?

**Pull request merged**